matlab/CMakeLists.txt

```diff
@@ -17,6 +17,23 @@
 
 cmake_minimum_required(VERSION 3.20)
 
+# This function uses the ThirdpartyToolchain CMake module
+# in cpp/cmake_modules to automatically download and build GoogleTest.
+function(download_and_build_google_test)
+  # The ExternalProject module is required for downloading and building external dependencies.
+  include(ExternalProject)
+  # The ThirdpartyToolchain module automatically builds xsimd by default,
+  # so we explicitly disable building of xsimd here.
+  set(ARROW_SIMD_LEVEL "NONE")
+  set(ARROW_RUNTIME_SIMD_LEVEL "NONE")
+  # The ThirdpartyToolchain module must know the location of the Arrow C++ libraries source code.
+  set(arrow_SOURCE_DIR ${CMAKE_SOURCE_DIR}/../cpp)
+  # The ThirdpartyToolchain module uses ExternalProject to download GoogleTest.
+  include(ThirdpartyToolchain)
```

Comment:
`ThirdpartyToolchain isn't expected to be used by `include`.
Could you use `find_package(GTest REQUIRED CONFIG)` instead?

Reply:
Thanks for reviewing these changes, @kou!

Could you elaborate on why `ThirdpartyToolchain` isn't expected to be `include`d in other CMake files?

We intentionally wanted to reuse the robust infrastructure that the community took time to develop for finding 3rd party libraries in a flexible way, rather than reinventing the wheel here.

Reply:
`ThirdpartyToolchain` has many Apache Arrow C++ specific configurations like `configure_file()` you changed.
So you need to set some `ARROW_*` variables to `include` `ThirdpartyToolchain`.

If we want to reuse `ThirdpartyToolchain` in the MATLAB library, we should make `ThirdpartyToolchain` reusable and reuse rather than `set` some variables and `include`.

Reply:
See my comment below.

It sounds like depending on `ThirdpartyToolchain` may not be desirable without first refactoring it to make it more reusable. Given this fact, I'm fine with instead pursuing the approach of trying to automatically build the Arrow C++ libraries first, and then depending on the bundled GoogleTest binaries that are built when `ARROW_BUILD_TESTS` is set to `on`.